api/v1alpha3/machine_types.go

```diff
@@ -29,6 +29,14 @@ const (
 	// MachineControlPlaneLabelName is the label set on machines or related objects that are part of a control plane.
 	MachineControlPlaneLabelName = "cluster.x-k8s.io/control-plane"
 
+	// ConditionHealthCheckSucceeded is set on machines that have passed a healthcheck by the MachineHealthCheck controller.
+	// In the event that the health check fails it will be set to False.
+	ConditionHealthCheckSucceeded ConditionType = "HealthCheckSucceeded"
```

Comment:
Should we move this to condition_consts.go? 
@vincepri ^^

Reply:
I'm not opposed, although we need to define a standard. `ReadyCondition` (in condition_consts.go`) has the `Condition` part at the end, while this one has it at the beginning.

It's actually nice that specific conditions are defined in the same file as the types